src/simplewallet/simplewallet.cpp

```diff
@@ -1304,6 +1398,615 @@ bool simple_wallet::export_raw_multisig(const std::vector<std::string> &args)
+	  % tr("Message Type") % tr("Height") % tr("Message State");
+  tools::message_store& ms = m_wallet->get_message_store();
+  for (size_t i = 0; i < messages.size(); ++i)
+  {
+    const tools::message &m = messages[i];
+    const tools::coalition_member &member = ms.get_member(m.member_index);
+    bool highlight = (m.state == tools::message_state_ready_to_send) || (m.state == tools::message_state_waiting);
+    message_writer(m.direction == tools::message_direction_out ? console_color_green : console_color_magenta, highlight) <<
+	    boost::format("%4s %-6s %-40s %-25s %7s %-25s") %
+	    m.id %
+	    ms.message_direction_to_string(m.direction) %
+	    ms.member_to_string(member, 40) %
+	    ms.message_type_to_string(m.type) %
+	    m.wallet_height %
+	    ms.message_state_to_string(m.state);
+  }
+}
+
+void simple_wallet::ask_send_all_ready_messages()
+{
+  tools::message_store& ms = m_wallet->get_message_store();
+  std::vector<tools::message> ready_messages;
+  const std::vector<tools::message> &messages = ms.get_all_messages();
+  for (size_t i = 0; i < messages.size(); ++i)
+  {
+    const tools::message &m = messages[i];
+    if (m.state == tools::message_state_ready_to_send)
+    {
+      ready_messages.push_back(m);
+    }
+  }
+  if (ready_messages.size() != 0)
+  {
+    list_mms_messages(ready_messages);
+    if (user_confirms(tr("Send these messages now?")))
+    {
+      for (size_t i = 0; i < ready_messages.size(); ++i)
+      {
+	ms.send_message(ready_messages[i].id);
+	ms.set_message_processed_or_sent(ready_messages[i].id);
+      }
+    }
+  }
+}
+
+bool simple_wallet::get_message_from_arg(const std::string &arg, tools::message &m)
+{
+  tools::message_store& ms = m_wallet->get_message_store();
+  bool valid_id = false;
+  uint32_t id;
+  try
+  {
+    id = (uint32_t)(std::stoi(arg));
+    m = ms.get_message_by_id(id);
+    valid_id = true;
+  }
+  catch (...)
+  {     
+    fail_msg_writer() << tr("Invalid message id");
+  }
+  return valid_id;
+}
+
+void simple_wallet::mms_init(const std::vector<std::string> &args)
+{
+  // mms init <threshold>/<coalition_size> <own_transport_address> 
+  // Example: mms init 2/3 rbrunner7@getmonero.org
+  // For now, assume we still have the original Monero address available, before "make_multisig"
+  if (args.size() != 3)
+  {
+    fail_msg_writer() << tr("usage: mms init <threshold>/<coalition_size> <own_transport_address>");
+    return;
+  }
+  tools::message_store& ms = m_wallet->get_message_store();
+  if (ms.is_active())
+  {
+    if (!user_confirms(tr("The MMS is already initialized. Re-initialize by deleting all member info and messages?")))
+    {
+      return;
+    }
+  }
+  uint32_t threshold;
+  uint32_t coalition_size;
+  const std::string &mn = args[1];
+  bool mn_ok = (mn.length() == 3) && (mn[1] == '/');  // do not support numbers > 9 for now
+  if (mn_ok)
```

Comment:
It would be simpler to read and understand it you placed negative scenario with exit before